src/engine/sparqlExpressions/NumericUnaryExpressions.cpp

```diff
@@ -20,10 +23,38 @@ inline auto unaryNegate = [](TernaryBool a) {
   }
   AD_FAIL();
 };
-NARY_EXPRESSION(UnaryNegateExpression, 1,
-                FV<decltype(unaryNegate), EffectiveBooleanValueGetter>,
-                SET<SetOfIntervals::Complement>);
 
+template <typename NaryOperation>
+requires(isOperation<NaryOperation>)
+class UnaryNegateExpressionImpl : public NaryExpression<NaryOperation> {
+ public:
+  using NaryExpression<NaryOperation>::NaryExpression;
+
+  std::optional<std::vector<PrefilterExprVariablePair>>
+  getPrefilterExpressionForMetadata(
+      [[maybe_unused]] bool isNegated) const override {
+    AD_CORRECTNESS_CHECK(this->N == 1);
+    auto optExprVarVec =
+        this->getNthChild(0).value()->getPrefilterExpressionForMetadata(
+            !isNegated);
```

Comment:
The `!isNegated` is important here but not very visible. Comment this explicitly.